wagtail_newsletter/static/wagtail_newsletter/js/wagtail_newsletter.js

```diff
@@ -65,53 +49,66 @@ window.wagtail.app.register("wn-panel",
         this.sendButtonProgress.loadingValue = false;
       }
     }
-
-    send() {
-      this.sendButtonProgress.activate();
-      this.sendSubmitTarget.click();
-    }
   }
 );
 
 
-window.wagtail.app.register("wn-test",
+window.wagtail.app.register("wn-send",
   class extends window.StimulusModule.Controller {
+    static targets = [
+      "message",
+    ];
+
     get dialog() {
       return this.application.getControllerForElementAndIdentifier(
         this.element.closest("[data-controller=w-dialog]"), "w-dialog"
       );
     }
 
-    submit() {
-      const address = this.element.querySelector("input[name=email]").value;
-      this.dispatch("submit", { detail: { address } });
-      this.dialog.hide();
+    show(event) {
+      const { name, member_count } = event.detail;
+      this.messageTarget.textContent = `Sending campaign to ${name}, with ${member_count} recipients.`;
+      this.dialog.show();
     }
   }
 );
 
 
-window.wagtail.app.register("wn-send",
+window.wagtail.app.register("wn-submit",
   class extends window.StimulusModule.Controller {
     static targets = [
-      "message",
-    ];
+      "button",
+    ]
 
-    get dialog() {
-      return this.application.getControllerForElementAndIdentifier(
-        this.element.closest("[data-controller=w-dialog]"), "w-dialog"
-      );
+    get button() {
+      return this.hasButtonTarget ? this.buttonTarget : this.element;
     }
 
-    show(event) {
-      const { name, member_count } = event.detail;
-      this.messageTarget.textContent = `Sending campaign to ${name}, with ${member_count} recipients.`;
-      this.dialog.show();
+    enter() {
```

Comment:
Suggestion: name your actions based on what will happen when they're called, [per Stimulus' conventions](https://stimulus.hotwired.dev/reference/actions#naming-conventions).

```suggestion
    submit() {
```

or maybe

```suggestion
    delayedSubmit() {
```

I don't know why it needs to have a `setTimeout()`, but if the delay is important to know, maybe name it `delayedSubmit` – although the delay is negligible as we're using `0`, so maybe not.